test/performance/performance_test.go

```diff
@@ -0,0 +1,325 @@
+	t.Log("Creating namespaces to be used by kperf.")
+	namespaces = append(namespaces, nsName)
+	ns := &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: nsName,
+		},
+	}
+	_, err = clients.KubeClient.CoreV1().Namespaces().Create(context.Background(), ns, metav1.CreateOptions{})
+	if err != nil {
+		t.Fatalf("Failed to create namespace %q: %#v", nsName, err)
+	}
+	defer cleanupNamespaces(t, clients, namespaces)
+
+	t.Log("Generating services by kperf.")
+	p := &kperf.PerfParams{
+		ClientSet:            clients.KubeClient,
+		NewServingClient:     servingClientForKperf,
+		NewAutoscalingClient: autoscalingClientForKperf,
+		NewNetworkingClient:  networkingClientForKperf,
+	}
+	generateArgs := kperf.GenerateArgs{
+		Number:      servicesCount,
+		Interval:    10,
+		Batch:       10,
+		Concurrency: 10,
+		MinScale:    1,
+		MaxScale:    2,
+		Namespace:   nsName,
+		SvcPrefix:   ServiceNamePrefix,
+		CheckReady:  true,
+		Timeout:     60 * time.Second,
+	}
+	err = service.GenerateServices(p, generateArgs)
+	if err != nil {
+		t.Fatalf("Failed to execute kperf service generate command %#v", err)
+	}
+	defer cleanupServices(t, p, nsName)
+
+	t.Log("Waiting for Services.")
+	servingClientInNamespace, err := servingClientForNamespace(nsName)
+	if err != nil {
+		t.Fatalf("Failed to get serving client %#v", err)
+	}
+
+	objs := []*v1.Service{}
+	for i := 0; i < servicesCount; i++ {
+		svcName := fmt.Sprintf("%s-%d", ServiceNamePrefix, i)
+		if err := v1test.WaitForServiceState(servingClientInNamespace, fmt.Sprintf("%s-%d", ServiceNamePrefix, i), v1test.IsServiceReady, "ServiceIsReady"); err != nil {
+			t.Fatalf("Failed waiting for serving to get Ready %#v", err)
+		}
+		svc, err := servingClientInNamespace.Services.Get(ctx, svcName, metav1.GetOptions{})
+		if err != nil {
+			t.Fatalf("Failed to get service %#v", err)
+		}
+		objs = append(objs, svc)
+	}
+
+	t.Log("Wait for Services to scale down to 0.")
+	err = waitForScaleToZero(ctx, clients, objs)
+	if err != nil {
+		t.Fatalf("Failed to wait for services to scale down to 0 %#v", err)
+	}
+
+	t.Log("Scale and measure results by kperf.")
+	tmpFolder, err := os.MkdirTemp("", "kperfoutput")
+	if err != nil {
+		t.Fatalf("Failed creating temp folder %#v", err)
+	}
+	defer os.RemoveAll(tmpFolder)
+
+	scaleArgs := kperf.ScaleArgs{
+		SvcPrefix: ServiceNamePrefix,
+		Namespace: nsName,
+		SvcRange:  servicesRange,
+		Verbose:   true,
+		Output:    tmpFolder,
+	}
+
+	err = service.ScaleServicesUpFromZero(p, scaleArgs)
+	if err != nil {
+		t.Fatalf("Failed to scale fropm zero %#v", err)
+	}
+
+	result := kperf.ScaleResult{}
+	err = getResultJson(tmpFolder, &result)
+	if err != nil {
+		t.Fatalf("Failed to parse kperf service scale result %#v", err)
+	}
+	for _, s := range result.Measurment {
+		if s.ServiceLatency > 0.5 {
+			t.Fatal("kperf Services scale from zero took too long")
+		}
+	}
+}
+
+func waitForScaleToZero(ctx context.Context, client *test.Clients, objs []*v1.Service) error {
```

Comment:
This one could come handy: https://github.com/knative/serving/blob/main/test/e2e/e2e.go#L73

Reply:
its not exactly the same, will look into it